webproject/src/main/java/com/google/sps/dataManagers/OrderManager.java

```diff
@@ -46,9 +49,9 @@ public Order addOrder(String userEmail, String product, int quantity, double uni
         orderMapping.put("orderPrice", orderPriceString);
         orderMapping.put("roomId", roomId);
         orderMapping.put("userEmailRoomId", userEmailRoomId);
-
-        firebaseUtil.getOrdersReference().push()
-                .setValue(orderMapping, (databaseError, databaseReference) -> {
+        orderMapping.put("orderId", orderRef.getKey());
```

Comment:
what is this getKey getting? is it getting existing key or generate a new key?

Reply:
It's getting the key of the reference to which the new Order is being assigned to.